chains/solana/contracts/programs/ccip-router/src/instructions/v1/onramp.rs

```diff
@@ -275,521 +215,239 @@ pub fn ccip_send<'info>(
+        // https://solana.com/developers/courses/program-security/reinitialization-attacks#add-is-initialized-check
+        if nonce_counter_account.version == 0 {
+            // The authority must be the owner of the PDA, as it's their Public Key in the seed
+            // If the account is not initialized, initialize it
+            nonce_counter_account.version = 1;
+            nonce_counter_account.counter = 0;
         }
-        _ => Err(CcipRouterError::InvalidExtraArgsTag.into()),
-    }
-}
-
-fn parse_and_validate_svm_extra_args(
-    cfg: &DestChainConfig,
-    tag: [u8; 4],
-    data: &mut &[u8],
-    message_contains_tokens: bool,
-) -> Result<ProcessedExtraArgs> {
-    match u32::from_be_bytes(tag) {
-        SVM_EXTRA_ARGS_V1_TAG => {
-            let args = if data.is_empty() {
-                SVMExtraArgsV1::default_config(cfg)
-            } else {
-                SVMExtraArgsV1::deserialize(data)?
-            };
 
-            // token_receiver != 0 when tokens are present
-            // token_receiver == 0 when tokens are not present
-            let receiver_is_zero_address = args.token_receiver == [0; 32];
-            require!(
-                message_contains_tokens != receiver_is_zero_address,
-                CcipRouterError::InvalidTokenReceiver
-            );
-
-            Ok(ProcessedExtraArgs {
-                bytes: args.serialize_with_tag(),
-                gas_limit: args.compute_units as u128,
-                allow_out_of_order_execution: args.allow_out_of_order_execution,
-            })
+        // config.enforce_out_of_order checked in `validate_svm2any`
+        let mut final_nonce = 0;
+        if !allow_out_of_order_execution {
+            nonce_counter_account.counter = nonce_counter_account.counter.checked_add(1).unwrap();
+            final_nonce = nonce_counter_account.counter;
         }
-        _ => Err(CcipRouterError::InvalidExtraArgsTag.into()),
-    }
-}
-
-fn bump_nonce(
-    nonce_counter_account: &mut Account<Nonce>,
-    allow_out_of_order_execution: bool,
-) -> Result<u64> {
-    // Avoid Re-initialization attack as the account is init_if_needed
-    // https://solana.com/developers/courses/program-security/reinitialization-attacks#add-is-initialized-check
-    if nonce_counter_account.version == 0 {
-        // The authority must be the owner of the PDA, as it's their Public Key in the seed
-        // If the account is not initialized, initialize it
-        nonce_counter_account.version = 1;
-        nonce_counter_account.counter = 0;
+        Ok(final_nonce)
     }
 
-    // config.enforce_out_of_order checked in `validate_svm2any`
-    let mut final_nonce = 0;
-    if !allow_out_of_order_execution {
-        nonce_counter_account.counter = nonce_counter_account.counter.checked_add(1).unwrap();
-        final_nonce = nonce_counter_account.counter;
+    pub(super) fn hash(msg: &SVM2AnyRampMessage) -> [u8; 32] {
```

Comment:
Did the hash change or is just moving code?

Reply:
Just moved it to remove an entire submodule after most code belonged to fee quoter